### PR TITLE
Cleanup shell scripts

### PR DESCRIPTION
**Pull request by mpoullet**
- apply shellcheck and use Bash strict mode.
- use the same prolog in each shell script.
- avoid exporting variables when not needed.
- take advantage of git clean to cleanup.
- be less verbose with apt-get install.
- upgrade to gRPC v1.16.0 to address issue #35. 
- fix language spelling.

### CODE REVIEW

File: tests/build.sh [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #!/bin/bash
 set -e
-set -x
+set -u
+set -o pipefail
+#set -x
```
Comment (Fleker) [inline tests/build.sh]: I personally liked having `set -x` despite its verbosity as it allowed me to see easily where an error may originate in a CI system.

Comment (mpoullet) [reply tests/build.sh]: Ok. I've reverted these lines.

**Comment by Fleker** [comment]
Overall looks good. I'm going to run it through a CI server just to be safe.

**Comment by mpoullet** [comment]
How was the CI run?

**Comment by mpoullet** [comment]
I've triggered a CI build of this PR on AppVeyor using Ubuntu 18.04 and it works fine as you can see [here](https://github.com/mpoullet/assistant-sdk-cpp/tree/appveyor) and [there](https://ci.appveyor.com/project/mpoullet/assistant-sdk-cpp/branch/appveyor).

It works fine on my Ubuntu 16.04 too.

However I can reproduce your issue with Travis CI on Ubuntu 14.04 too, see [here](https://travis-ci.org/mpoullet/assistant-sdk-cpp/builds/451290059).

At the moment I'm stuck: gRPC v1.15.0 is needed to build the actual googleapis by providing protobuf v3.6.1 to avoid #35. gRPC v1.15.0 works on Ubuntu 18.04/16.04 but not 14.04.

gRPC v1.12.0/v1.13.0/v1.14.0 are of no help as they all build protobuf version 3.5.2.

I'd like to know how you can build `tests/build.sh` as is without hitting #35.

Possible actions:
- get gRPC v1.15.0 running on Ubuntu 14.04 -> best solution but how?
- remove commit 0f6e02a8689e6c6e0703084e496fa44e4d5a52b3, but then `tests/build.sh` should hit #35 -> do you see this on your CI too?
- stick to gRPC v1.11.0 but then checkout a specific commit from googleapis that still works with protobuf v3.5.0 OR delete the `google/ads/`directory in `tests/build.sh`?

**Comment by mpoullet** [comment]
I've narrowed/reproduced/reported the gRPC issue [here](https://github.com/grpc/grpc/issues/17136). You might ask some fellow coworkers to have a look at it.

**Comment by Fleker** [comment]
Yeah if we revert back to 1.11.0, then we can't compile the protobufs correctly, unless we only select a subset of the files to compile. It would be okay to [`make` here](https://github.com/googlesamples/assistant-sdk-cpp/blob/master/tests/build.sh#L42) and pass in a list of files to make. This would also speed up the process since we aren't actually using most.

Then, we can go back to v1.15.0 later once issues there are resolved.

**Comment by mpoullet** [comment]
It's not really an issue with `gRPC` but rather with `protobuf` `v3.6.0` in gRPC. I've updated the gRPC [issue](https://github.com/grpc/grpc/issues/17136) accordingly. Another `git bisect` should find the `protobuf` issue but I need to find time to investigate further.

**Pull request closed without merging**